common/src/chain/transaction/signature/inputsig.rs

```diff
@@ -88,6 +88,12 @@ impl StandardInputSignature {
                 verify_public_key_spending(pubkey, &sig_components, sighash)?
             }
             Destination::ScriptHash(_) => return Err(TransactionSigError::Unsupported),
+            Destination::AnyoneCanSpend => {
+                // AnyoneCanSpend must use InputWitness::NoSignature, so this is unreachable
```

Comment:
wouldn't `unreachable!()` more appropriate in that case?

Reply:
Lukas asked the same question about something else. I'm not confident someone won't by mistake call these for some reason. I'm afraid that using expect and unreachable in certain complex places can lead to attack vectors.

Reply:
> Lukas asked the same question about something else. I'm not confident someone won't by mistake call these for some reason. I'm afraid that using expect and unreachable in certain complex places can lead to attack vectors.

While I understand the concern, the tradeoff is that it will be more difficult to detect bugs, which is not great either. Seems there is no good option here.

Reply:
Well, detecting bugs can be done through proper logging after all. It's not ideal, but it's the best we got.